core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -171,12 +173,15 @@ private[spark] class CoarseMesosSchedulerBackend(
     synchronized {
       val filters = Filters.newBuilder().setRefuseSeconds(5).build()
 
-      for (offer <- offers) {
+      // filter out all the offers that do not meet constraints (if specified)
+      val qualifyingOffers = filterOffersByConstraints(offers, slaveOfferConstraints)
```

Comment:
I might be wrong, but doesn't this mean that this framework misbehaves in that it does not decline offers that doesn't match the constraints? These are left hanging until they time out.